### PR TITLE
Authentication/login

### PR DESCRIPTION
**Pull request by MuhammadUsama-hub**

### CODE REVIEW

File: src/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -105,4 +119,19 @@ export class AuthController extends Controller {
 
     return toResponse({ data: "Email successfully verified!" });
   }
-}
+
+  @Delete("/signoutAll")
```
Comment (MohammedMaaz) [inline src/modules/auth/auth.controller.ts]: This method should be POST

File: src/modules/auth/auth.controller.ts [hunk 2]
```diff
@@ -105,4 +119,19 @@ export class AuthController extends Controller {
...
+  @Middlewares(validateAuthentication)
+  public async signout(@Request() req: ExReq): Promise<APIResponse<string>> {
+      const user = getReqUser(req)
+      //refresh token version changed
```
Comment (MohammedMaaz) [inline src/modules/auth/auth.controller.ts]: the following code is self explanatory, so no need for the comment

File: prisma/schema.prisma [hunk 3]
```diff
@@ -26,23 +26,24 @@ model posts {
...
-  id String@unique(map: "_pgroll_new_users_xata_id_key") @default(dbgenerated("('rec_'::text || (xata_private.xid())::text)")) @map("xata_id")
+  created_at DateTime @default(now()) @map("xata_createdat") @db.Timestamptz(6)
+  updated_at DateTime @default(now()) @map("xata_updatedat") @db.Timestamptz(6)
+  id String   @unique(map: "_pgroll_new_users_xata_id_key") @default(dbgenerated("('rec_'::text || (xata_private.xid())::text)")) @map("xata_id")
   refresh_token_version Int      @default(0)
   role String   @default("user")
   email_verified Boolean  @default(false)
   password_hash String
   email String   @unique(map: "users_email_unique")
+  refresh_tokens String?  @default("\"\"")
```
Comment (MohammedMaaz) [inline prisma/schema.prisma]: refresh_tokens are not needed now, `yarn dev` should fix this

**Review by MohammedMaaz — CHANGES_REQUESTED** [review]
@MuhammadUsama-hub good work overall!

Added few minor comments

File: src/modules/auth/auth.controller.ts (hunk 1 above)
Comment (MuhammadUsama-hub) [reply src/modules/auth/auth.controller.ts]: ok

File: src/modules/auth/auth.controller.ts (hunk 2 above)
Comment (MuhammadUsama-hub) [reply src/modules/auth/auth.controller.ts]: ok

File: prisma/schema.prisma (hunk 3 above)
Comment (MuhammadUsama-hub) [reply prisma/schema.prisma]: ok

**Comment by MuhammadUsama-hub** [comment]
Successfully Edit  all comments

**Comment by MuhammadUsama-hub** [comment]
Successfully Edit all comments.

**Pull request closed without merging**